src/lib/sqlite/sqliterg.test.ts

```diff
@@ -0,0 +1,460 @@
+import { describe, it, expect, expectTypeOf, beforeEach, beforeAll } from 'vitest'
+import {
+  createRemoteDb,
+  createRemoteFoldersStore,
+  createRemoteSnippetStore,
+  createSqlitergExecutor,
+  migrateRemote
+} from './sqliterg'
+import type { SqlitergExecutor, ResponseExecuteError } from './sqliterg'
+import { migrate, defaultMigrationQueryMap, defaultQueriesStringMap } from './migration'
+import type { MigrationState } from './migration'
+import { sql } from 'drizzle-orm'
+import { get, writable } from 'svelte/store'
+import { snippets as snippets_, folders } from '$lib/sqlite/schema'
+import type { Folder, GlobalState, RemoteFoldersStore } from '$lib/utitlities/persistence';
+import { createNewSnippet } from '$lib/utitlities/persistence'
+import { waitUntil } from '$lib/utitlities/wait-until';
+import {
+  deleteAllSnippets,
+  deleteFolder,
+  querySnippetsByFolderId,
+  upsertFolder,
+  upsertSnippet
+} from '$lib/sqlite/queries';
+import { createDummySnippet } from '$lib/utitlities/testing'
+import type { SqliteRemoteDatabase } from 'drizzle-orm/sqlite-proxy';
+import type { DisplayFolder } from '$lib/utitlities/data-transformation';
+
+enum Constants {
+  ServerURL = 'http://127.0.0.1:12321/crypta',
+  ServerURLWrong = 'http://127.0.0.1:12322/crypta',
+  Blank = '',
+  Username = 'crypta',
+  Password = 'crypta',
+  PasswordWrong = 'wrong'
+}
+
+export function createEmptyExecutor(): SqlitergExecutor {
+  return createSqlitergExecutor(
+    Constants.Blank,
+    Constants.Blank,
+    Constants.Blank,
+  )
+}
+
+export function createUnreachableExecutor(): SqlitergExecutor {
+  return createSqlitergExecutor(
+    Constants.ServerURLWrong,
+    Constants.Blank,
+    Constants.Blank,
+  )
+}
+
+function createUnauthenticatedExecutor(): SqlitergExecutor {
+  return createSqlitergExecutor(
+    Constants.ServerURL,
+    Constants.Username,
+    Constants.PasswordWrong,
+  )
+}
+
+export function createAvailableExecutor(): SqlitergExecutor {
+  return createSqlitergExecutor(
+    Constants.ServerURL,
+    Constants.Username,
+    Constants.Password,
+  )
+}
+
+// IMPORTANT: `yarn dev-db` should be run before the tests are executed
+describe('executor', () => {
```

Comment:
TODO: investigate how to start a `ws4sqlite` process on test setting up and stop the process on test tearing down (#37).